service/src/main/scala/com/intuit/superglue/lineage/LineageService.scala

```diff
@@ -33,16 +33,28 @@ class LineageService(val repository: SuperglueRepository) {
       case _ =>
     }
 
-    val futureGraph = for {
-      table <- repository.tableRepository.getByName(tableName)
-      tablePK = table.head.id
-    } yield tableLineage(Set(tablePK), Set(tablePK), Graph.empty, backwardDepth, forwardDepth)
-    futureGraph.flatten.map { case Graph(nodes, links) =>
-      val annotatedNodes = nodes.map {
-        case TableNode(pk, name, _) if name.toLowerCase == tableName.toLowerCase => TableNode(pk, name, "selected")
-        case other => other
+    if (backwardDepth.contains(0) && forwardDepth.contains(0)) {
```

Comment:
What if backwardDepth is 0 but forwardDepth > 0 ? Have we tested that scenarion ?

Reply:
The condition I have added here affects only the flow where bw=0&fw=0. Originally it used to go the traversal route and no nodes will be found because both depths are 0. And so it was returning a table not found error. So, added the check to avoid traversal if depths are 0 and return the table node itself.
If one of the depths is >0, it goes in the traversal flow and creates a Graph. See below

> http://localhost:8080/api/v1/lineage/table/RUTRUM?bw=1&fw=0

{"nodes":[{"id":3452156367131786611,"group":"table","label":"ELIT"},{"id":1315625410024507986,"group":"table","label":"ADIPISCING"},{"id":-1439824252242955195,"group":"table","label":"DOLOR"},{"id":4708353957227732056,"group":"selected","label":"RUTRUM"},{"id":7822254195634786844,"group":"table","label":"IPSUM"},{"id":-295374950813635157,"group":"table","label":"FRINGILLA"},{"id":8076363984277974215,"group":"table","label":"SIT"},{"id":-4943664676160088363,"group":"table","label":"VIVAMUS"},{"id":2186611429998809561,"group":"table","label":"CONSECTETUR"},{"id":6469050193342992248,"group":"table","label":"LOREM"}],"edges":[{"from":1315625410024507986,"to":4708353957227732056},{"from":-1439824252242955195,"to":-4943664676160088363},{"from":8076363984277974215,"to":1315625410024507986},{"from":6469050193342992248,"to":2186611429998809561},{"from":8076363984277974215,"to":2186611429998809561},{"from":3452156367131786611,"to":4708353957227732056},{"from":4708353957227732056,"to":-295374950813635157},{"from":7822254195634786844,"to":-4943664676160088363},{"from":2186611429998809561,"to":4708353957227732056},{"from":-4943664676160088363,"to":4708353957227732056},{"from":8076363984277974215,"to":-4943664676160088363}]}


> http://localhost:8080/api/v1/lineage/table/RUTRUM?bw=0&fw=1

{"nodes":[{"id":3452156367131786611,"group":"table","label":"ELIT"},{"id":1315625410024507986,"group":"table","label":"ADIPISCING"},{"id":-1439824252242955195,"group":"table","label":"DOLOR"},{"id":4708353957227732056,"group":"selected","label":"RUTRUM"},{"id":7822254195634786844,"group":"table","label":"IPSUM"},{"id":-295374950813635157,"group":"table","label":"FRINGILLA"},{"id":8076363984277974215,"group":"table","label":"SIT"},{"id":-4943664676160088363,"group":"table","label":"VIVAMUS"},{"id":2186611429998809561,"group":"table","label":"CONSECTETUR"},{"id":6469050193342992248,"group":"table","label":"LOREM"}],"edges":[{"from":1315625410024507986,"to":4708353957227732056},{"from":-1439824252242955195,"to":-4943664676160088363},{"from":8076363984277974215,"to":1315625410024507986},{"from":6469050193342992248,"to":2186611429998809561},{"from":8076363984277974215,"to":2186611429998809561},{"from":3452156367131786611,"to":4708353957227732056},{"from":4708353957227732056,"to":-295374950813635157},{"from":7822254195634786844,"to":-4943664676160088363},{"from":2186611429998809561,"to":4708353957227732056},{"from":-4943664676160088363,"to":4708353957227732056},{"from":8076363984277974215,"to":-4943664676160088363}]}

Reply:
Got it. Thanks for clarifying.